ml-prims/src/cuda_utils.h

```diff
@@ -347,9 +347,10 @@ HDI double myPow(double x, double power) {
  * @defgroup Pow Power function
  * @{
  */
-template <typename Type>
+// IdxType mostly to be used for MainLambda in *Reduction kernels
+template <typename Type, typename IdxType = int>
 struct Nop {
-  HDI Type operator()(Type in) { return in; }
+  HDI Type operator()(Type in, IdxType i = 0) { return in; }
```

Comment:
@teju85 We'd need both the `MainNop` and a `Nop` since only the `main_op` requires an index passed in and `final_op` does not have the index as an argument. Maybe call it `Nop` and `ArgNop`?

Reply:
I was looking for an easier way to reuse the same name `Nop` for all our cases. The default template param for `IdxType` over the class and the default arg for `i` in this method should also just work, no?

Reply:
Yes, that should work. Sorry, I missed the default template argument.

Other than that, all changes LGTM!